src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -55,13 +71,24 @@ public CommandResult execute(Model model) {
         Predicate<Person> combinedPredicate = predicates.stream()
                 .reduce(Predicate::and)
                 .orElse(t -> false);
+
+        if (isChained) {
+            // We HAVE to copy the list here, or it will have been emptied by the time Predicate.test() is called
+            Predicate<Person> inFilteredListPredicate = new InFilteredListPredicate(
+                    new ArrayList<>(model.getFilteredPersonList()));
+            combinedPredicate = combinedPredicate.and(inFilteredListPredicate);
+        }
+
         model.updateFilteredPersonList(combinedPredicate);
-        String formattedKeywords = getAllKeywordsFromPredicates(predicates);
+        String conditions = getAllKeywordsFromPredicates(predicates);
+        if (isChained) {
+            conditions = conditions + ", among the previously displayed results";
```

Comment:
Would it be better to reformat the string in a different way and extract the string part out as a variable to avoid the issue of magic literal? For example, `In the current filtered list: 1 person(s) found with condition: (c)` and extract the `In the current filtered list: ` as a separated variable in `Messages.java` with name such as `ChainedFindResultPrefix` etc?